src/uart.rs

```diff
@@ -441,8 +441,8 @@ impl Device {
     }
 
     fn reset<'a>(self) -> &'a mut ConfigMmio {
-        const NULL: *mut ConfigMmio = core::ptr::null_mut();
-        let uart = unsafe { &mut *NULL.with_addr(self.addr()) };
+        let regs = core::ptr::from_exposed_addr_mut::<ConfigMmio>(self.addr());
+        let uart = unsafe { &mut *regs };
```

Comment:
The pointer from `from_exposed_addr(something.addr())`  would be UB if this were an address inside the AM. And this pattern in general is very lintable and trips the linter between my ears. I'd suggest you change if it I could come up with a better name, but at some point a new nightly clippy might be upset by this.

And maybe soon we will have `offset_of!` in order to do this address computation for the `write_volatile` in a way that for sure doesn't have the aliasing implications of `&mut`. I hope.